### ISSUE TITLE
ALLOW_MEMORY_GROWTH not working with USE_PTHREADS=1

### ISSUE THREAD

**Issue by thomasjammet** [kripken/emscripten#7110]
I don't know if this behavior is expected but every time I set USE_PTHREADS=1 ALLOW_MEMORY_GROWTH=1 and WASM_MEM_MAX are ignored.

You can reproduce the issue by compiling [test_memorygrowth_wasm_mem_max.c](https://github.com/kripken/emscripten/blob/incoming/tests/core/test_memorygrowth_wasm_mem_max.c) with the following command :

```
emcc .\test_memorygrowth_wasm_mem_max.c -s USE_PTHREADS=1 -s TOTAL_ME
MORY=64Mb -s ALLOW_MEMORY_GROWTH=1 -s WASM_MEM_MAX=100Mb -o testmem.html
```

**Comment by MarcelKlammer** [comment]
preamble.js generates this:

```
function enlargeMemory() {
  abort('Cannot enlarge memory arrays, since compiling with pthreads support enabled (-s USE_PTHREADS=1).');
}
```

**Comment by dschuff** [comment]
I'm guessing this is an intentional limitation left over from before wasm threads support was added (i.e. when only asm.js was supported). For wasm, resizing is allowed; the [requirement](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#resizing) is that the max memory size must be specified (instead of being left unspecified, which places no limit on growth up to 4GB). Probably I just never got around to making that behavior correct when I added the initial wasm threads support. So `WASM_MEM_MAX` should work, and we should should require it to be specified if `ALLOW_MEMORY_GROWTH=1`. I (or someone) will hopefully fix this soon, otherwise I'd be happy to review a PR :)
